spec/takeaway_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'takeaway'
+require 'current_order'
+
+describe Takeaway do
+
+  TEST_MEAL = 'Mega Breakfast'
+  TEST_ORDER = { item: "Mega Breakfast", price: 6.75 }
+  let(:send_message) { double }
+
+  describe '#menu' do
+    it 'returns a list of menu' do
+      expect(subject.menu).to eq(subject.menu)
+    end
+  end
+
+  describe '#choose_meal' do
+    it 'selects a meal from the menu' do
+      expect(subject.choose_meal(TEST_MEAL)).to eq([TEST_ORDER])
+    end
+  end
+
+  describe '#confirm_order' do
+    it 'Confirms the order and gives the correct price.' do
+      subject.choose_meal(TEST_MEAL)
+      subject.choose_meal(TEST_MEAL)
+      expect(subject.confirm_order).to eq(13.5)
+    end
+  end
+
+  describe '#pay_for_order' do
+    it 'Clears a customers order and sends a text' do
+      allow(subject).to receive(:send_message) { nil }
+      subject.choose_meal(TEST_MEAL)
+      subject.choose_meal(TEST_MEAL)
+      expect(subject.pay_for_order).to eq([])
+    end
+  end
+end
```

Comment:
So, the hardest part of this exercise is testing the text message sending stuff, which you've side-stepped by making it a private method :)

I'll give you some tips on that anyway...

```
client.messages.create(
      from: from,
      to: to,
      body: "Thank you! Your order was placed and will be delivered before #{Time.new.hour + 1}:#{Time.new.min}"
      )
```

So you have a `client` that responds to `messages` and returns _something_ that responds to `create`.  To make this work, you need to be calling `create` with some sensible params.  Here's how you might create some doubles for this test...

```
let(:messages)  { double :twilio_messages }
let(:client)          { double :twilio_client, messages: messages }
```
Then in your test, you'd use a stub to declare that you expect `messages` to receive `create` with the correct params...

```
expect(messages).to receive(:create).with(
      from: from,
      to: to,
      body: "Thank you! Your order was placed and will be delivered before #{Time.new.hour + 1}:#{Time.new.min}"
)
```